catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPreStoragePlugin.java

```diff
@@ -238,38 +300,75 @@ private ContentItem createDerivedImage(String id, String qualifier, BufferedImag
             byte[] overviewBytes = scaleImage(image, maxWidth, maxHeight);
 
             ByteSource source = ByteSource.wrap(overviewBytes);
-            ContentItem contentItem = new ContentItemImpl(id, qualifier, source, IMAGE_JPEG,
-                    buildDerivedImageTitle(metacard.getTitle(), qualifier), overviewBytes.length,
+            ContentItem contentItem = new ContentItemImpl(id,
+                    qualifier,
+                    source,
+                    IMAGE_JPEG,
+                    buildDerivedImageTitle(metacard.getTitle(), qualifier, JPG),
+                    overviewBytes.length,
+                    metacard);
+
+            addDerivedResourceAttribute(metacard, contentItem);
+
+            return contentItem;
+        } catch (IOException e) {
+            LOGGER.debug(e.getMessage(), e);
+        }
+
+        return null;
+    }
+
+    private ContentItem createOriginalImage(String id, BufferedImage image, Metacard metacard) {
+
+        try {
+            byte[] originalBytes = renderToJpeg2k(image);
+
+            ByteSource source = ByteSource.wrap(originalBytes);
+            ContentItem contentItem = new ContentItemImpl(id,
+                    ORIGINAL,
+                    source,
+                    IMAGE_JPEG2K,
+                    buildDerivedImageTitle(metacard.getTitle(), ORIGINAL, JP2),
+                    originalBytes.length,
                     metacard);
 
             addDerivedResourceAttribute(metacard, contentItem);
 
             return contentItem;
+
         } catch (IOException e) {
             LOGGER.debug(e.getMessage(), e);
         }
 
         return null;
     }
 
-    String buildDerivedImageTitle(String title, String qualifier) {
+    private String buildDerivedImageTitle(String title, String qualifier, String extension) {
         String rootFileName = FilenameUtils.getBaseName(title);
 
         // title must contain some alphanumeric, human readable characters, or use default filename
-        if (StringUtils.isNotBlank(rootFileName)
-                && StringUtils.isNotBlank(rootFileName.replaceAll("[^A-Za-z0-9]", ""))) {
+        if (StringUtils.isNotBlank(rootFileName) && StringUtils.isNotBlank(rootFileName.replaceAll(
```

Comment:
I tried your approach, but I had trouble getting the existing unit tests to pass. It looks like it is because of underscores. The first regex does not include underscore, by the replaceAll does.